File: apps/neuronav_app/lib/core/constants/app_constants.dart
/// Core application constants for NeuroNav.
///
/// Ported from Swift AppConstants to Dart/Flutter.
library;

// ---------------------------------------------------------------------------
// Supabase
// ---------------------------------------------------------------------------

const String supabaseUrl = 'https://fornsbnwtrorqkmfabri.supabase.co';

const String supabaseAnonKey =
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.'
    'eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImZvcm5zYm53dHJvcnFrbWZhYnJpIiw'
    'icm9sZSI6ImFub24iLCJpYXQiOjE3NzA5MTQ1MjIsImV4cCI6MjA4NjQ5MDUyMn0.'
    'kp3ywqW-4S4M8tysNRGLLVgBSXQkwRswPRefEuRp_u8';

// ---------------------------------------------------------------------------
// Platform identifiers
// ---------------------------------------------------------------------------

const String appGroupIdentifier = 'group.com.neuronav.shared';
const String keychainService = 'com.neuronav.keychain';

// ---------------------------------------------------------------------------
// UserRole
// ---------------------------------------------------------------------------

enum UserRole {
  guest,
  patient,
  caregiver,
  family,
  admin;

  /// Human-readable name in Spanish.
  String get displayName {
    switch (this) {
      case UserRole.guest:
        return 'Invitado';
      case UserRole.patient:
        return 'Paciente';
      case UserRole.caregiver:
        return 'Cuidador';
      case UserRole.family:
        return 'Familiar';
      case UserRole.admin:
        return 'Administrador';
    }
  }

  /// Whether the role is allowed to create / edit / delete data.
  bool get canModifyData {
    switch (this) {
      case UserRole.guest:
        return false;
      case UserRole.patient:
        return true;
      case UserRole.caregiver:
        return true;
      case UserRole.family:
        return false;
      case UserRole.admin:
        return true;
    }
  }

  /// Build a [UserRole] from a raw string coming from the backend.
  ///
  /// The backend may send `"user"` which maps to [UserRole.patient].
  /// Unknown strings default to [UserRole.guest].
  factory UserRole.fromString(String value) {
    switch (value.toLowerCase()) {
      case 'guest':
        return UserRole.guest;
      case 'patient':
      case 'user':
        return UserRole.patient;
      case 'caregiver':
        return UserRole.caregiver;
      case 'family':
        return UserRole.family;
      case 'admin':
        return UserRole.admin;
      default:
        return UserRole.guest;
    }
  }
}

// ---------------------------------------------------------------------------
// RoutineCategory
// ---------------------------------------------------------------------------

enum RoutineCategory {
  cooking,
  hygiene,
  laundry,
  medication,
  transit,
  shopping,
  cleaning,
  social,
  custom;

  /// Material Icons name suitable for use with [Icons] or icon fonts.
  String get icon {
    switch (this) {
      case RoutineCategory.cooking:
        return 'restaurant';
      case RoutineCategory.hygiene:
        return 'shower';
      case RoutineCategory.laundry:
        return 'local_laundry_service';
      case RoutineCategory.medication:
        return 'medication';
      case RoutineCategory.transit:
        return 'directions_bus';
      case RoutineCategory.shopping:
        return 'shopping_cart';
      case RoutineCategory.cleaning:
        return 'cleaning_services';
      case RoutineCategory.social:
        return 'people';
      case RoutineCategory.custom:
        return 'tune';
    }
  }

  /// Human-readable label in Spanish.
  String get displayName {
    switch (this) {
      case RoutineCategory.cooking:
        return 'Cocina';
      case RoutineCategory.hygiene:
        return 'Higiene';
      case RoutineCategory.laundry:
        return 'Lavanderia';
      case RoutineCategory.medication:
        return 'Medicacion';
      case RoutineCategory.transit:
        return 'Transporte';
      case RoutineCategory.shopping:
        return 'Compras';
      case RoutineCategory.cleaning:
        return 'Limpieza';
      case RoutineCategory.social:
        return 'Social';
      case RoutineCategory.custom:
        return 'Personalizada';
    }
  }

  /// Build a [RoutineCategory] from a raw backend string.
  factory RoutineCategory.fromString(String value) {
    switch (value.toLowerCase()) {
      case 'cooking':
        return RoutineCategory.cooking;
      case 'hygiene':
        return RoutineCategory.hygiene;
      case 'laundry':
        return RoutineCategory.laundry;
      case 'medication':
        return RoutineCategory.medication;
      case 'transit':
        return RoutineCategory.transit;
      case 'shopping':
        return RoutineCategory.shopping;
      case 'cleaning':
        return RoutineCategory.cleaning;
      case 'social':
        return RoutineCategory.social;
      case 'custom':
      default:
        return RoutineCategory.custom;
    }
  }
}

// ---------------------------------------------------------------------------
// ScheduleType
// ---------------------------------------------------------------------------

enum ScheduleType {
  daily,
  weekdays,
  weekends,
  specific,
  once;

  String get value {
    return name;
  }

  factory ScheduleType.fromString(String value) {
    switch (value.toLowerCase()) {
      case 'daily':
        return ScheduleType.daily;
      case 'weekdays':
        return ScheduleType.weekdays;
      case 'weekends':
        return ScheduleType.weekends;
      case 'specific':
        return ScheduleType.specific;
      case 'once':
        return ScheduleType.once;
      default:
        return ScheduleType.daily;
    }
  }
}

// ---------------------------------------------------------------------------
// ExecutionStatus
// ---------------------------------------------------------------------------

enum ExecutionStatus {
  inProgress,
  completed,
  aborted,
  paused;

  /// The raw string sent to / received from the backend.
  String get value {
    switch (this) {
      case ExecutionStatus.inProgress:
        return 'in_progress';
      case ExecutionStatus.completed:
        return 'completed';
      case ExecutionStatus.aborted:
        return 'aborted';
      case ExecutionStatus.paused:
        return 'paused';
    }
  }

  factory ExecutionStatus.fromString(String value) {
    switch (value.toLowerCase()) {
      case 'in_progress':
        return ExecutionStatus.inProgress;
      case 'completed':
        return ExecutionStatus.completed;
      case 'aborted':
        return ExecutionStatus.aborted;
      case 'paused':
        return ExecutionStatus.paused;
      default:
        return ExecutionStatus.inProgress;
    }
  }
}

// ---------------------------------------------------------------------------
// StepExecutionStatus
// ---------------------------------------------------------------------------

enum StepExecutionStatus {
  pending,
  inProgress,
  completed,
  skipped,
  needsHelp;

  /// The raw string sent to / received from the backend.
  String get value {
    switch (this) {
      case StepExecutionStatus.pending:
        return 'pending';
      case StepExecutionStatus.inProgress:
        return 'in_progress';
      case StepExecutionStatus.completed:
        return 'completed';
      case StepExecutionStatus.skipped:
        return 'skipped';
      case StepExecutionStatus.needsHelp:
        return 'needs_help';
    }
  }

  factory StepExecutionStatus.fromString(String value) {
    switch (value.toLowerCase()) {
      case 'pending':
        return StepExecutionStatus.pending;
      case 'in_progress':
        return StepExecutionStatus.inProgress;
      case 'completed':
        return StepExecutionStatus.completed;
      case 'skipped':
        return StepExecutionStatus.skipped;
      case 'needs_help':
        return StepExecutionStatus.needsHelp;
      default:
        return StepExecutionStatus.pending;
    }
  }
}

File: apps/ios/AdaptAiKit/Sources/Constants/AppConstants.swift
import Foundation


public enum AppConstants {
    public static let supabaseURL = "https://hrfipfmxbdaoipjcszif.supabase.co"
    public static let supabaseAnonKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhyZmlwZm14YmRhb2lwamNzemlmIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NzI2NDI5ODIsImV4cCI6MjA4ODIxODk4Mn0.DnA7b9yXOCtPSC0beXOxa7NCMMFssaZZBtA0FsVlG_0"

    public static let appGroupIdentifier = "group.com.adaptai.shared"
    public static let keychainService = "com.adaptai.keychain"

    public enum UserRole: String, CaseIterable {
        case guest
        case patient
        case caregiver
        case family
        case admin

        // Map old "user" value from DB
        public init(rawValue: String) {
            switch rawValue {
            case "guest": self = .guest
            case "patient", "user": self = .patient
            case "caregiver": self = .caregiver
            case "family": self = .family
            case "admin": self = .admin
            default: self = .patient
            }
        }

        public var displayName: String {
            switch self {
            case .guest: return "Invitado"
            case .patient: return "Paciente"
            case .caregiver: return "Cuidador"
            case .family: return "Familiar"
            case .admin: return "Admin"
            }
        }

        public var canModifyData: Bool {
            switch self {
            case .guest, .family: return false
            case .patient, .caregiver, .admin: return true
            }
        }
    }

    public enum RoutineCategory: String, CaseIterable {
        case cooking
        case hygiene
        case laundry
        case medication
        case transit
        case shopping
        case cleaning
        case social
        case custom

        public var icon: String {
            switch self {
            case .cooking: return "fork.knife"
            case .hygiene: return "shower.fill"
            case .laundry: return "washer.fill"
            case .medication: return "pill.fill"
            case .transit: return "bus.fill"
            case .shopping: return "cart.fill"
            case .cleaning: return "bubbles.and.sparkles.fill"
            case .social: return "person.2.fill"
            case .custom: return "star.fill"
            }
        }
    }

    public enum ScheduleType: String, CaseIterable {
        case daily
        case weekdays
        case weekends
        case custom
    }

    public enum ExecutionStatus: String, Codable {
        case inProgress = "in_progress"
        case completed
        case paused
        case abandoned
    }

    public enum StepExecutionStatus: String, Codable {
        case pending
        case inProgress = "in_progress"
        case completed
        case skipped
        case error
    }

    public enum LinkStatus: String, Codable {
        case pending
        case active
        case revoked
    }

    public enum AlertSeverity: String, Codable {
        case low
        case medium
        case high
        case critical
    }

    public enum AlertType: String, Codable {
        case emergency
        case zoneExit = "zone_exit"
        case zoneEnter = "zone_enter"
        case medicationMissed = "medication_missed"
        case stallDetected = "stall_detected"
        case crashDetected = "crash_detected"
        case system
    }

    public enum AppointmentStatus: String, Codable {
        case scheduled
        case completed
        case cancelled
    }
}
